### DISCUSSION TITLE
WHY Use IMCOMPLETE+HASH As Part Of The Transferring File Title?

### DISCUSSION THREAD

**Discussion by DeathStalker77** [nicotine-plus/nicotine-plus#2682] [Q&A]
This seems to be completely unnecessary, and also one cannot tell from which user the file originates.  Why not simply use the same methodology that the original SLSK used?  Incomplete files go in the User's (or other identified) folder until complete?  Adding long extensions like this can create problems with file/path length limitations.

Additionally, why do ALL of a selected user's files have to be completed, before they are moved to the "completed" folder?  Why not simply do it folder by folder?

Lastly (for now), what happens when there is an existing file with the same name?

Thanks!

--- DS

(I'm working on transitioning from the original SLSK to N+, but am encountering a LOT of issues - I can assist with ideas, processes, & logic - I am a certified PSPO II, so this IS what I do for a living, not a hobby).

**Comment by mathiascode** [comment]
> This seems to be completely unnecessary, and also one cannot tell from which user the file originates. Why not simply use the same methodology that the original SLSK used?

Because someone decided to use this format 20 years ago. It was based on the format the original Soulseek client (Soulseek NS) used, and there weren't good enough reasons to go through the trouble of migrating to a new one.

> Adding long extensions like this can create problems with file/path length limitations.

It's known, and was resolved eventually: https://github.com/nicotine-plus/nicotine-plus/pull/2053

> Additionally, why do ALL of a selected user's files have to be completed, before they are moved to the "completed" folder? Why not simply do it folder by folder?

There's probably an issue moving the files to your completed download folder (judging by your post on Reddit). A file is supposed to be moved out of the incomplete folder immediately after it finishes.

> Lastly (for now), what happens when there is an existing file with the same name?

Answered on Reddit: https://www.reddit.com/r/Soulseek/comments/175hrou/slsk_n_running_together_related_issues/k4kznfb/

> (I'm working on transitioning from the original SLSK to N+, but am encountering a LOT of issues - I can assist with ideas, processes, & logic - I am a certified PSPO II, so this IS what I do for a living, not a hobby).

The best thing to do is to open individual issues here, then we can look into them: https://github.com/nicotine-plus/nicotine-plus/issues

Before submitting feature requests, please check that a request hasn't been made previously. Thanks.

**Reply by DeathStalker77** [reply]
_Because someone decided to use this format 20 years ago. It was based on the format the original Soulseek client (Soulseek NS) used, and there weren't good enough reasons to go through the trouble of migrating to a new one._

Ok, but that certainly seems like something that SHOULD be addressed now.  I can understand it being used at the start, to get out an MVP, but at this point, impo, it should be re-revaluated - **KISS Principle**

_Adding long extensions like this can create problems with file/path length limitations.
It's known, and was resolved eventually: https://github.com/nicotine-plus/nicotine-plus/pull/2053_

Well it created a problem for me when it tried to move the file to the completed folder (where the total path WAS longer than 255).  I had to manually move it elsewhere.  So it seems that maybe this IS still an issue that should be looked into :-)

_There's probably an issue moving the files to your completed download folder (judging by your post on Reddit). A file is supposed to be moved out of the incomplete folder immediately after it finishes._

If that is the case, then what is there to do about it?

And yes, I saw the answer to my last query answered :-)

--- DS

**Reply by mathiascode** [reply]
Can you send full logs containing the error messages to my email address (on my GitHub profile)?

Instructions are here (enable the downloads, uploads, transfers, miscellaneous debug categories): https://nicotine-plus.org/doc/DEVELOPING.html#debug-logging
